channeldb/paginate.go

```diff
@@ -78,6 +78,14 @@ func (p paginater) cursorStart() ([]byte, []byte) {
 			indexKey, indexValue = p.keyValueForIndex(
 				p.indexOffset - 1,
 			)
+
+			// If the key and value one before our index is nil, the
+			// index offset we have been provided is beyond our last
+			// item. Since we are paginating backwards, we simply
+			// start at the last item.
+			if indexKey == nil {
```

Comment:
Not part of test case?

Reply:
This behaviour is pre-existing for our invoices pagination, so I wanted to separate it out. The previous commit moves the logic across as is, and this one improves it. But I think a better way would be to add the failing test then fix it, going to switch the order around.